libheif/CMakeLists.txt

```diff
@@ -1,84 +1,106 @@
-set (libheif_sources
-bitstream.h
-bitstream.cc
-box.cc
-box.h
-error.cc
-error.h
-heif_api_structs.h
-heif.cc
-heif_context.cc
-heif_context.h
-heif_file.cc
-heif_file.h
-heif.h
-heif_image.cc
-heif_image.h
-heif_hevc.h
-heif_hevc.cc
-heif_colorconversion.h
-heif_colorconversion.cc
-heif_plugin_registry.h
-heif_plugin_registry.cc
-heif_limits.h
-heif_plugin.h
-heif_plugin.cc
-heif_version.h
-logging.h
-)
+include(CMakePackageConfigHelpers)
 
 if(UNIX)
   include (${CMAKE_ROOT}/Modules/FindPkgConfig.cmake)
   pkg_check_modules (LIBDE265 libde265)
   pkg_check_modules (X265 x265)
 endif()
 
-if(LIBDE265_FOUND)
-  add_definitions(-DHAVE_LIBDE265=1)
-  set (libheif_sources
-    ${libheif_sources}
-    heif_decoder_libde265.cc
-    heif_decoder_libde265.h
-  )
-  set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} ${LIBDE265_CFLAGS}")
-  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${LIBDE265_CFLAGS}")
-
-  if (NOT "${LIBDE265_LIBRARY_DIRS}" STREQUAL "")
-    set(LIBDE265_LINKDIR "-L${LIBDE265_LIBRARY_DIRS}")
-  endif()
-endif()
+set(libheif_headers
+    bitstream.h
+    box.h
+    error.h
+    heif_api_structs.h
+    heif_context.h
+    heif_file.h
+    heif.h
+    heif_image.h
+    heif_hevc.h
+    heif_colorconversion.h
+    heif_plugin_registry.h
+    heif_limits.h
+    heif_plugin.h
+    logging.h
+    ${PROJECT_BINARY_DIR}/libheif/heif_version.h
+)
 
-if(X265_FOUND)
-  add_definitions(-DHAVE_X265=1)
-  set (libheif_sources
-    ${libheif_sources}
-    heif_encoder_x265.cc
-    heif_encoder_x265.h
-  )
-  set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} ${X265_CFLAGS}")
-  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${X265_CFLAGS}")
-
-  if (NOT "${X265_LIBRARY_DIRS}" STREQUAL "")
-    set(X265_LINKDIR "-L${X265_LIBRARY_DIRS}")
-  endif()
-endif()
+add_library(${PROJECT_NAME} SHARED
+    bitstream.cc
+    box.cc
+    error.cc
+    heif.cc
+    heif_context.cc
+    heif_file.cc
+    heif_image.cc
+    heif_hevc.cc
+    heif_colorconversion.cc
+    heif_plugin_registry.cc
+    heif_plugin.cc
+    ${libheif_headers}
+)
+
+# Needed to find libheif/heif_version.h while compiling the library
+target_include_directories(${PROJECT_NAME} PRIVATE ${PROJECT_BINARY_DIR})
 
-# set(CMAKE_VERBOSE_MAKEFILE on)
+# Propagate include/libheif to consume the headers from other projects
+target_include_directories(${PROJECT_NAME} PUBLIC
+    $<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}/include/${PROJECT_NAME}>
```

Comment:
Will this add both `include` and `include/libheif` to the exported include path? It should be sufficient to add just `include` as we want other apps to explicitly include using `#include <libheif/heif.h>` (instead of just `#include <heif.h>`).

Reply:
Good catch. I copy-pasted-modified this snippet from other project where I needed to do that and I forgot to cleanup here.